terraform/aks/storage_account.tf

```diff
@@ -0,0 +1,8 @@
+resource "azurerm_storage_account" "sa" {
+  count                    = var.create_storage_account ? 1 : 0
+  name                     = var.storage_account_name
+  resource_group_name      = local.app_resource_group_name
+  location                 = "UK South"
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
```

Comment:
make sure we have soft delete for the container and sa, although maybe just for prod